From 2d8179e6e60f175da3e5ea8c9e527d438d80af7c Mon Sep 17 00:00:00 2001
From: Emmanuele Bassi <ebassi@gnome.org>
Date: Fri, 30 Oct 2015 11:49:31 +0000
Subject: [PATCH] Use better vendor strings from ATi/AMD

Improve the detection of ATi/AMD GPUs, especially latter ones.

https://bugzilla.gnome.org/show_bug.cgi?id=756306
---
 cogl/cogl-gpu-info.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/cogl/cogl-gpu-info.c b/cogl/cogl-gpu-info.c
index 54dfe18..e153a99 100644
--- a/cogl/cogl-gpu-info.c
+++ b/cogl/cogl-gpu-info.c
@@ -178,10 +178,12 @@ check_nvidia_vendor (const CoglGpuInfoStrings *strings)
 static CoglBool
 check_ati_vendor (const CoglGpuInfoStrings *strings)
 {
-  if (strcmp (strings->vendor_string, "ATI") != 0)
-    return FALSE;
+  if (strcmp (strings->vendor_string, "ATI Technologies Inc.") == 0)
+    return TRUE;
+  if (strcmp (strings->vendor_string, "Advanced Micro Devices, Inc.") == 0)
+    return TRUE;
 
-  return TRUE;
+  return FALSE;
 }
 
 static CoglBool
-- 
2.5.0